### PR TITLE
Feature lidar setup

### PR DESCRIPTION
**Pull request by JanezCim**
Added LS lidar instructions to learn pages. This is a draft pull request because more discussion is yet to be done:

Unresolved questions:

- Since we are likely to add the LD lidar, we need to make sure we have the tutorials prepared for that. Now the biggest issue is that i still dont know how we are going to be calling this lidar in the tutorials. Internally we define them as LS and LD lidars, but since we want to be branding the LS lidar as our own, what the heck do we call it in the tutorials? I've refered to it as magni-default lidar or ubiquity robotics lidar? 
- The same question for little LS lidar - are we branding it as UR lidar? If yes, how do we call it.

Please rewiew the tutorials and give me feedback. Thanks!

### CODE REVIEW

File: doing_more/ls_lidar_setup_kinetic.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+layout: default
+title:  "Setting up magni-default lidar Kinetic"
+permalink: default_lidar_setup
+---
+# Setting up ubiquity robotics default lidar - Kinetic
+
+#### &uarr;[top]( https://ubiquityrobotics.github.io/learn/)
+
+Ubuntu 18.04
```
Comment (MoffKalast) [inline doing_more/ls_lidar_setup_kinetic.md:10]: 16.04?

**Review by MoffKalast — CHANGES_REQUESTED** [review]
Found a few issues to resolve, also we should probably discuss the best way to launch lidars on our images in a more seamless fashion.

File: doing_more/ls_lidar_setup_common.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+---
+layout: default
+title:  "Setting up magni-default lidar"
+permalink: default_lidar_setup
+---
+# Setting up ubiquity robotics default lidar
+
+#### &uarr;[top]( https://ubiquityrobotics.github.io/learn/)
+
+In this document it is described how to set up the LS301 lidar on the Magni.
```
[the remark this answers is not in the document]
Comment (JanezCim) [reply doing_more/ls_lidar_setup_common.md:10]: which.... is also not bad :D

yes, will fix this

File: doing_more/ls_lidar_setup_common.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
+---
+layout: default
+title:  "Setting up magni-default lidar"
+permalink: default_lidar_setup
+---
+# Setting up ubiquity robotics default lidar
```
[the remark this answers is not in the document]
Comment (JanezCim) [reply doing_more/ls_lidar_setup_common.md:6]: Ok, true, do you have any suggestion as to what to call it? Im fine with LSN301 and LD06 but we all need to agree on calling it that then. See description of PR for more comments about naming?

File: doing_more/ls_lidar_setup_common.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    cd ~/catkin_ws/
+    catkin_make
+    
+
+# Running
+
+    cd ~/catkin_ws/
+    source devel/setup.bash
+    roslaunch lslidar_n301_decoder lslidar_n301_config.launch device_IP:=192.168.42.222
+
```
[the remark this answers is not in the document]
Comment (JanezCim) [reply doing_more/ls_lidar_setup_common.md:65]: yeah if those lidars will become default on magni, maybe the launches could be in magni repo as well.

File: doing_more/ls_lidar_setup_common.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+Default camera and lidar positions can be found [here](https://github.com/UbiquityRobotics/magni_robot/tree/noetic-devel/magni_description/extrinsics)
+
+# Setup to work with RaspberryPi
+
+## Network
+The lidar should come pre-configured with static IP: 192.168.42.222 and only answering to requests coming from IP 192.168.42.125, so we need to configure that on the RPI.
+
+A static interface needs to be set on RPIs lan port by editing the interfaces file
+
+    sudo nano /etc/network/interfaces
```
[the remark this answers is not in the document]
Comment (JanezCim) [reply doing_more/ls_lidar_setup_common.md:42]: Ah yes, im now finding ways to make this work also on focal, Untill then i've just added a todo in a comment

File: doing_more/ls_lidar_setup_common.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (MoffKalast) [reply doing_more/ls_lidar_setup_common.md:6]: Right I missed that part of the description, I suppose LS-N301 and LD06 is good. Full names would be LeiShen N301 and LDRobot LD06.

**Comment by MoffKalast** [comment]
> You mean that on the magni_robot repo, noetic branch? Any idea how to make it more seamless? Id be happy to hear it. :)

Yeah I think it may be worthwhile to have it as a yaml param much like sonars_installed, except lidar_type or something similar. Much like it's done [in ezmap kinetic](https://github.com/UbiquityRobotics/ezmap/blob/8a0bd219d84acf260377a768f7384a94147ea0a1/ezmap_bringup/launch/core.launch#L101) right now for the time being.

**Comment by JanezCim** [comment]
@MoffKalast @mjstn i've now renamed the ls lidar into ur50 lidar in all the tutorials. i've also implemented @MoffKalast s suggestions. I think its now ready to be reviewed. I think we think about adding tutorials for other lidars in other PRs

**Comment by JanezCim** [comment]
@MoffKalast and @mjstn this has been stale for some time now but its ready to be reviewed and merged if all is alright.

**Review by MoffKalast — APPROVED** [review]
Ok yeah the changes seem good for now, looks alright to me. There will always be a few more issues found once it's hosted, so we can everything that's noted later.

**Pull request merged**